velox/common/memory/SharedArbitrator.cpp

```diff
@@ -225,28 +227,26 @@ bool SharedArbitrator::ensureCapacity(
   return checkCapacityGrowth(*requestor, targetBytes);
 }
 
-void SharedArbitrator::handleOOM(
+bool SharedArbitrator::handleOOM(
     MemoryPool* requestor,
     uint64_t targetBytes,
     std::vector<Candidate>& candidates) {
   MemoryPool* victim =
       findCandidateWithLargestCapacity(requestor, targetBytes, candidates).pool;
-  if (requestor != victim) {
-    VELOX_MEM_LOG(WARNING) << "Aborting victim memory pool " << victim->name()
-                           << " to free up memory for requestor "
-                           << requestor->name();
-  } else {
-    VELOX_MEM_LOG(ERROR) << "Aborting requestor memory pool "
-                         << requestor->name() << " itself to free up memory";
+  if (requestor == victim) {
+    VELOX_MEM_LOG(ERROR)
+        << "Requestor memory pool " << requestor->name()
+        << " is selected as victim memory pool so fail the memory arbitration";
+    return false;
   }
+  VELOX_MEM_LOG(WARNING) << "Aborting victim memory pool " << victim->name()
+                         << " to free up memory for requestor "
+                         << requestor->name();
   abort(victim);
   // Free up all the unused capacity from the aborted memory pool and gives back
   // to the arbitrator.
   incrementFreeCapacity(victim->shrink());
-  if (requestor == victim) {
-    ++numFailures_;
-    VELOX_MEM_POOL_ABORTED(requestor);
-  }
+  return true;
```

Comment:
Along with the issue description
> Also skip aborting requestor if the memory pool is selected as victim
memory pool to abort when arbitration fails. For some memory
arbitration use case like maybeReserve call, the user is able to handle
the arbitration failure by spilling itself.

Wasn't the caller already able to "spill itself" since the caller pool was included in `candidates` of call to `arbitrateMemory`? Did the change mean that we are going to add a spilling method that is more aggressive comparing to the current ways?

Reply:
No, this might not be needed as we are improving the abort error message. In most cases, the arbitrator is able to reclaim from an operator under maybeReserve call.